contributors/devel/owners.md

```diff
@@ -1,100 +1,228 @@
+- [munger:
+  blunderbuss](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/blunderbuss.go)
+  - responsible for determining **reviewers** and assigning to them
+  - chooses from people in the deepest/closest OWNERS files to the code being changed
+  - weights its choice based on the magnitude of lines changed for each file
+  - randomly chooses to ensure the same people aren't chosen every time
+  - if its `blunderbuss-number-assignees` flag is unset, it will default to 2 assignees
+- [munger:
+  approval-handler](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/approval-handler.go)
+  - responsible for adding the  `approved` label once an **approver** for each of the required
+    OWNERS files has `/approve`'d
+  - responsible for commenting as required OWNERS files are satisfied
+  - responsible for removing outdated approval status comments
+  - [full description of the
+    algorithm](https://github.com/kubernetes/test-infra/blob/6f5df70c29528db89d07106a8156411068518cbc/mungegithub/mungers/approval-handler.go#L99-L111)
+- [munger:
+  submit-queue](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/submit-queue.go) 
+  - responsible for merging PR's<D-d><D-d>
+  - responsible for updating a GitHub status check explaining why a PR can't be merged (eg: a
+    missing `lgtm` or `approved` label)
+
+### [`prow`](https://github.com/kubernetes/test-infra/tree/master/prow)
+
+Prow receives events from GitHub, and reacts to them. It is effectively stateless.
+
+- [plugin: lgtm](https://github.com/kubernetes/test-infra/tree/master/prow/plugins/lgtm)
+  - responsible for adding the `lgtm` label when a **reviewer** comments `/lgtm` on a PR
+  - the **PR author** may not `/lgtm` their own PR
+- [plugin: assign](https://github.com/kubernetes/test-infra/tree/master/prow/plugins/assign)
+  - responsible for assigning github users in response to `/assign` comments on a PR
+  - responsible for unassigning github users in response to `/unsassign` comments on a PR
+
+### GitHub
+
+GitHub provides a few integration points for our automation to do its thing.
+
+- [Status Checks](https://help.github.com/articles/about-required-status-checks/): our tests and
+  submit queue use these to communicate whether a commit is OK
+- [Protected Branches](https://help.github.com/articles/about-protected-branches/): ensure that a
+  branch cannot be merged unless all status checks are green
+- [Merge Button](https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button): the
+  submit queue effectively uses an API driven click of this button
+
+
+## Updating OWNERS
+
+OWNERS files should be regularly maintained.
+
+We should strive to:
+
+- grow the number of OWNERS files
+- add new people to OWNERS files
+- keep people in OWNERS files who are actively contributing to or reviewing the code they own
+- remove inactive people from OWNERS files
+
```

Comment:
We should also try to get rid of all non-collaborators in OWNERS files. When the bots try to assign issues or PRs to those people they fail. The OWNERS files should really only contain members and outside collaborators.